test/test_mimic_explainer.py

```diff
@@ -339,6 +346,46 @@ def test_explain_model_imbalanced_classes(self, mimic_explainer):
         assert np.isclose(surrogate_predictions, model_predictions).all()
         assert global_explanation.method == LIGHTGBM_METHOD
 
+    def test_explain_model_string_classes(self, mimic_explainer):
+        adult_census_income = retrieve_dataset('AdultCensusIncome.csv', skipinitialspace=True)
+        X = adult_census_income.drop(['income'], axis=1)
+        y = adult_census_income[['income']]
+        features = X.columns.values.tolist()
+        classes = y['income'].unique().tolist()
+        pipe_cfg = {
+            'num_cols' : X.dtypes[X.dtypes=='int64'].index.values.tolist(),
+            'cat_cols' : X.dtypes[X.dtypes=='object'].index.values.tolist(),
+        }
+        num_pipe = Pipeline([
+            ('num_imputer', SimpleImputer(strategy='median')),
+            ('num_scaler', StandardScaler())
+        ])
+        cat_pipe = Pipeline([
+            ('cat_imputer', SimpleImputer(strategy='constant', fill_value='?')),
+            ('cat_encoder', OneHotEncoder(handle_unknown='ignore', sparse=False))
+        ])
+        feat_pipe = ColumnTransformer([
+            ('num_pipe', num_pipe, pipe_cfg['num_cols']),
+            ('cat_pipe', cat_pipe, pipe_cfg['cat_cols'])
+        ])
+        X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, test_size=0.1)
```

Comment:
X_test, y_test are never used below. Why split the dataset and do other operations on them?

Reply:
Don't you need X_test for transform as well.

Reply:
good point -- actually most of this code came directly from the user with minor modifications, I didn't realize some of the variables were not used

Reply:
removed